src/app/_components/Footer.js

```diff
@@ -1,12 +1,21 @@
+'use client';
+
 import Link from 'next/link';
 import { WhiteLogo } from '@icons';
 import { cn } from '@utils/cn';
-import siteNav from '@config/siteNav';
 import { PillButton } from '@components/PillButton';
-import { SocialLink } from '@components/Links';
+import { SocialLinksList } from '@components/Links';
+import { useEffect, useState } from 'react';
+import { getSocialMediaList } from '@components/Links/socialMediaActions';
 
 export function Footer() {
-  const { links } = siteNav;
+  const [applicationLink, setApplicationLink] = useState('');
+
+  useEffect(() => {
+    getSocialMediaList().then(data => {
+      setApplicationLink(data.applicationLink);
+    });
```

Comment:
Can we transform footer.js to a server component and get the links before we render?

Reply:
Yeah, we can do that, thanks!